pkg/portal/cluster/machines_test.go

```diff
@@ -97,3 +113,133 @@ func TestMachines(t *testing.T) {
+				{
+					Kind: func(v string) *string { return &v }("something"),
+					Type: func(v string) *string { return &v }("Microsoft.Compute/virtualMachines"),
+					Name: func(v string) *string { return &v }("master-x"),
+				},
+			},
+			returnedVM: mgmtcompute.VirtualMachine{
+				Name: func(v string) *string { return &v }("master-x"),
+				VirtualMachineProperties: &mgmtcompute.VirtualMachineProperties{
+					InstanceView: &mgmtcompute.VirtualMachineInstanceView{
+						Statuses: &[]mgmtcompute.InstanceViewStatus{
+							{
+								Code: func() *string {
+									s := new(string)
+									*s = "PowerState/running"
+									return s
+								}(),
+							},
+						},
+					},
+				},
+			},
+			wantOutput: map[string]string{"master-x": "PowerState/running"},
+			wantErr:    nil,
+		},
+		{
+			name:                            "No VM resource found",
+			returnedRefreshableAuthorizer:   mockRefreshable,
+			returnedGenericResourceExpanded: []mgmtfeatures.GenericResourceExpanded{},
+			wantOutput:                      map[string]string{},
+			wantErr:                         nil,
+		},
+		{
+			name:                            "Empty FP Authorizer",
+			returnedRefreshableAuthorizer:   nil,
+			returnedGenericResourceExpanded: nil,
+			wantOutput:                      nil,
+			wantErr:                         errors.New("Empty Athorizer"),
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			mockEnv.EXPECT().Environment().Return(&azureclient.AROEnvironment{
+				Environment: azure.Environment{
+					ResourceManagerEndpoint: "temp",
+				},
+			})
+			mockEnv.EXPECT().FPAuthorizer(gomock.Any(), gomock.Any()).Return(tt.returnedRefreshableAuthorizer, tt.wantErr)
+			if tt.returnedRefreshableAuthorizer != nil {
+				mockResourcesClient.EXPECT().ListByResourceGroup(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).
+					Return(tt.returnedGenericResourceExpanded, tt.wantErr)
+			}
+			if len(tt.returnedGenericResourceExpanded) > 0 {
+				mockVirtualMachinesClient.EXPECT().Get(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(tt.returnedVM, tt.wantErr)
+			}
+			_, log := testlog.New()
+			mcf := newMockClientFactory(mockResourcesClient, mockVirtualMachinesClient)
+			azureSideFetcher := &azureSideFetcher{
+				log:                          log,
+				resourceGroupName:            "someResourceGroup",
+				subscriptionDoc:              subscriptionDoc,
+				env:                          mockEnv,
+				resourceClientFactory:        mcf,
+				virtualMachinesClientFactory: mcf,
+			}
+			azureClient := &azureClient{fetcher: azureSideFetcher, log: log}
+			vmAllocationStatuses, err := azureClient.VMAllocationStatus(ctx)
+			if !reflect.DeepEqual(vmAllocationStatuses, tt.wantOutput) {
+				t.Error("Expected output", tt.wantOutput, "Got", vmAllocationStatuses)
+			}
+			if err != nil && err /*.Error()*/ != tt.wantErr || err == nil && tt.wantErr != nil /*""*/ {
```

Comment:
left over comments

Reply:
removed these leftover comments.
And yes the bug which Ellis pointed out, I fixed it in this PR.